### ISSUE TITLE
Documentation

### ISSUE THREAD

**Issue by pepordev** [bropat/eufy-security-client#1]
Hi! thanks for the amazing work, do you have plans to add any documentation/examples on how to use the client?

**Comment by bropat** [comment]
Hi @pedroparraortega,
yes I have, but at the moment the further development of the client has priority. But you can see in [this project](https://github.com/bropat/ioBroker.eufy-security) how I use it.

**Comment by maltic42** [comment]
Hi, I need a small script that puts my hub into deactivated mode or away mode. Unfortunately I cannot get the 2fa working. Anyone who has a small example for me on using the library?

**Comment by xraywinedrinker** [comment]
> Hi, I need a small script that puts my hub into deactivated mode or away mode. Unfortunately I cannot get the 2fa working. Anyone who has a small example for me on using the library?

I second this. I was able to use the eufy node client example from Jan Loebel but don’t know how to create or modify for bropat’s eufy security client. I used the p2p index.js file in python flask and then trigger mode changes in webcore/smartthings.

**Comment by fbenniks** [comment]
Hi, to change arming mode i use the following code snippet, it took me a couple of hours to browse the code with some trail and error but this works on an eufy account without 2FA.

process.env.eufyBaseStationSN; can be found using 'await eufy.getStations(); 

`async setGuardState(state) {
    const { GuardMode } = require('eufy-security-client/build/http/types');
    const { EufySecurity } = require('eufy-security-client');

    let armMode = GuardMode.DISARMED;
    if (state === global.constants.HOME.HOME) {
      armMode = GuardMode.HOME;
    } else if (state === global.constants.HOME.AWAY) {
      armMode = GuardMode.AWAY;
    }

    const config = {
      country: 'nl',
      username: process.env.eufyUsername,
      password: process.env.eufyPassword,
    };
    const stationSN = process.env.eufyBaseStationSN;
    const eufy = new EufySecurity(config);
    let resultMessage = 'resultMessage';
    if (await eufy.connect()) {
      await eufy.refreshData();
      return new Promise((resolve) => {
        eufy
          .getStation(stationSN) //get SN from use let stations = await eufy.getStations()
          .on('connect', (station) => {
            let currentMode = station.getGuardMode().value;
            if (currentMode === armMode) {
              resultMessage = 'Nothing to change, eufy arm mode already:' + state;
              station.close();
            } else {
              station.setGuardMode(armMode);
            }
          })
          .on('guard mode', (station) => {
            let currentModeInt = station.getGuardMode().value;
            let currentModeName = GuardMode[currentModeInt];
            resultMessage = 'guard mode changed to:' + currentModeName;
            console.log(resultMessage);
            station.close();
          })
          .on('close', () => {
            resolve(resultMessage);
          });
      });
    }
  }`

i Hope this accelerates someones project since it took me a couple of hours to try to understand how the npm is working:)

Greetz

**Comment by fuatakgun** [comment]
Hey @bropat , can you include home assistant python integration into examples section?

**Comment by azavisha** [comment]
@bropat Can you please explain the difference between `Device.updateProperty` and `Device.setParameters`? It looks like only `setParameters` can be used to actually change some device settings. If that is the case, what is the purpose of `updateProperty` if it doesn't do anything useful?

**Comment by jacobhweeks** [comment]
How do I get started with this after npm install eufy-security-client?

**Comment by jaredkotoff** [comment]
> process.env.eufyBaseStationSN; can be found using `await eufy.getStations(); ` after running `await eufy.refreshData();` getStations() returns an empty array if refreshData() has not been run:)

Was `refreshData()` removed? Im using node-red wrapper for this project and I get an empty array when trying to get list of devices. I saw the refresh data command but throws an error because refreshData function is not defined.

**Comment by bropat** [comment]
@jaredkotoff 

> Update: nvm see that it was renamed to refreshCloudData

Exactly, but I see that this project uses a [very old version](https://github.com/recallfx/node-red-contrib-eufy-security/blob/master/package.json#L25) of my library (`1.0.0` is about 2 years old). Current version: `2.7.0`.
I strongly recommend updating the library.

**Comment by jaredkotoff** [comment]
Yep, I'm working on a PR!

**Comment by fbelkadhi** [comment]
Dear Sir @bropat hello, no one could deny the importance of your unique and magnificent works, we can never thank you enough for that, thank you very much by the way.
On the other hand, what is the point of creating an extraordinary product like **[eufy.security.client](https://github.com/bropat/eufy-security-client)** without providing documentation on how to configure it from A to Z and ignoring people who ask you for it?!
Personally, I am an intermediate user of Homebridge and I really want to make considerable donations but how can I give for one or more products whose creator does not even allow me to use?
It would be immensely appreciated by all of you if you could allow us to use your excellent creation.
Thank you very much.

**Comment by max246** [comment]
> Dear Sir [@bropat](https://github.com/bropat) hello, no one could deny the importance of your unique and magnificent works, we can never thank you enough for that, thank you very much by the way. On the other hand, what is the point of creating an extraordinary product like **[eufy.security.client](https://github.com/bropat/eufy-security-client)** without providing documentation on how to configure it from A to Z and ignoring people who ask you for it?! Personally, I am an intermediate user of Homebridge and I really want to make considerable donations but how can I give for one or more products whose creator does not even allow me to use? It would be immensely appreciated by all of you if you could allow us to use your excellent creation. Thank you very much.

Thank you for your consideration, did we go anywhere from this? We can work on some documentation since many are missing.
If you have any pr in mind, please open it and I will be happy to review it

**Comment by max246** [comment]
> PR added to fix this [#717](https://github.com/bropat/eufy-security-client/pull/717)

Will have a look when i am back from holiday but thanks
